contribute/README.md

```diff
@@ -10,9 +10,13 @@ Thanks for enhancing OpenFisca anyway!
 
 ## Why contribute to OpenFisca?
 
-OpenFisca is a project being developed under the GPLv3 license or later.
+OpenFisca is a project being developed under the AGPL-3.0 license.
 The source code is freely available and modifiable.
 
+We ask for improvements and evolution of OpenFisca to be shared back, for it allows the community to grow through improvements to the _core engine_ and updates to the _modelled tax-benefit systems_ (country packages, extensions, reforms, etc.).
+
+Although, as long as the software you build with OpenFisca can't be considered an improvement or an evolution of OpenFisca, we won't consider it a form of derivative work. If the latter holds true, we won't ever ask you to share your software's source code, whether you're using the Python API, the Web API or both.
```

Comment:
> What do we mean by `OpenFisca`? Could a ui like legislation explorer be considered an evolution of OpenFisca?

Reply:
Good question : a modelling of the legislation and a way to execute it given some input ?

Reply:
@MattiSG you may have thoughts about all this :).

Reply:
I find this set of provisions confusing. I'm quite aligned with @fpagnoux's description, and I don't retrieve it as clearly in the added paragraph.

I see the point of a very detailed intentions explanation, along with archetypical examples that make the threat model explicit (e.g. a bank trying to free-ride on legislations to make cross-country comparisons as competitive advantage is a big no). I see the point of a license that is legally binding. I am not certain to see the point of this intermediary presentation that opens flexibility while still leaving plenty of room for interpretation (what is constitutive of an “improvement”?).

@maukoquiroga could you make the intention behind this changeset clearer? Whose and which needs are we trying to solve with this paragraph? 🙂

Reply:
> As a developper of france, I don't want third parties to free-ride on the model without sharing back their improvements

Of course !

>  I'm not quite as sure, but I'd also say that you should share back your code

I think it's technically difficult not to do so, as you said I could create a big `extension` and a small `country` and do not share it back. I think virality should extend to `extensions` and `reforms`.

> If a group or organisation models a new country, should they be allowed to use whatever license they want, as they are the one who did the modeling effort ?

I think licence should apply here also.

> If OpenFisca were used to model a country that doesn't exist (for research purpose, games, etc.), I'd see this a similar to the potential reform use-case, and be less strict about it.

Like a game, for example. I do agree, although again it could be really hard to make the difference and add an exceptions. Unless you have a wording idea ?

> I see the point of this intermediary presentation that opens flexibility while still leaving plenty of room for interpretation (what is constitutive of an “improvement”?).

> @maukoquiroga could you make the intention behind this changeset clearer? Whose and which needs are we trying to solve with this paragraph?

I want to know what to tell companies that are using OpenFisca, whether or not they have to open their application code.

Personally, I think now, but for all the rest, I think virality should apply. I'd like to achieve [this](https://redislabs.com/blog/why-redis-labs-modules-are-agpl/). I'll add that to the description of this pull request.